framework/src/ble_manager/ble_manager_autoconnect.c

```diff
@@ -0,0 +1,249 @@
+	while (ctx->state == BLE_CLIENT_AUTOCONNECTING) {
+		if (state == BLE_AUTOCON_STATE_DISCONNECT) {
+			ret = ble_client_start_scan(NULL, &scan_config);
+			if (ret != BLE_MANAGER_SUCCESS) {
+				BLE_LOG_ERROR("[BLEMGR] auto conn scan fail[%d]\n", ret);
+				goto finish_auto;
+			}
+		}
+		memset(buf, 0, sizeof(buf));
+		nbytes = mq_receive(ctx->recv_mqfd, buf, sizeof(buf), 0);
+		evt = (ble_autocon_event)(buf[0]);
+		BLE_LOG_DEBUG("[BLEMGR] mqueue evt : %d\n", evt);
+
+		/* This is finishing process */
+		if (evt == BLE_AUTOCON_EVT_ALERT || conn_stop == 1) {
+			conn_stop = 1;
+			if (state == BLE_AUTOCON_STATE_SCAN_STARTING) {
+				state = BLE_AUTOCON_STATE_SCAN_STOPPING;
+				ret = ble_client_stop_scan();
+				if (ret != BLE_MANAGER_SUCCESS) {
+					BLE_LOG_ERROR("[BLEMGR] auto conn scan stop fail[%d]\n", ret);
+					goto finish_auto;
+				}
+			} else if (state == BLE_AUTOCON_STATE_SCAN_STOPPING) {
+				if (evt == BLE_AUTOCON_EVT_SCAN_STOP) {
+					state = BLE_AUTOCON_STATE_ALERT;
+				}
+			} else if (state == BLE_AUTOCON_STATE_CONNECTING) {
+				if (evt == BLE_AUTOCON_EVT_CONNECTED) {
+					state = BLE_AUTOCON_STATE_CONNECTED;
+					ctx->auto_connect = false;
+					break;
+				} else if (evt == BLE_AUTOCON_EVT_DISCONNECTED) {
+					state = BLE_AUTOCON_STATE_ALERT;
+				}
+			} else {
+				state = BLE_AUTOCON_STATE_ALERT;
+			}
+
+			if (state == BLE_AUTOCON_STATE_ALERT) {
+				ctx->state = BLE_CLIENT_IDLE;
+				ctx->auto_connect = false;
+				break;
+			}
+
+		/* This is normal process */
+		} else if (state == BLE_AUTOCON_STATE_DISCONNECT) {
+			if (evt == BLE_AUTOCON_EVT_SCAN_START) {
+				state = BLE_AUTOCON_STATE_SCAN_STARTING;
+			}
+		} else if (state == BLE_AUTOCON_STATE_SCAN_STARTING) {
+			if (evt == BLE_AUTOCON_EVT_MAC_SCANNED) {
+				state = BLE_AUTOCON_STATE_SCAN_STOPPING;
+				ret = ble_client_stop_scan();
+				if (ret != BLE_MANAGER_SUCCESS) {
+					BLE_LOG_ERROR("[BLEMGR] auto conn scan stop fail[%d]\n", ret);
+					goto finish_auto;
+				}
+			}
+		} else if (state == BLE_AUTOCON_STATE_SCAN_STOPPING) {
+			if (evt == BLE_AUTOCON_EVT_SCAN_STOP) {
+				state = BLE_AUTOCON_STATE_CONNECTING;
+				ret = ble_client_reconnect(ctx);
+				if (ret != BLE_MANAGER_SUCCESS) {
+					BLE_LOG_ERROR("[BLEMGR] auto conn reconnect fail[%d]\n", ret);
+					goto finish_auto;
+				}
+			}
+		} else if (state == BLE_AUTOCON_STATE_CONNECTING) {
+			if (evt == BLE_AUTOCON_EVT_CONNECTED) {
+				state = BLE_AUTOCON_STATE_CONNECTED;
+				break;
+			} else if (evt == BLE_AUTOCON_EVT_DISCONNECTED) {
+				state = BLE_AUTOCON_STATE_DISCONNECT;
+			}
+		}
+		BLE_LOG_DEBUG("[BLEMGR] auto conn state : %d\n", state);
+	}
+finish_auto:
+	mq_close(ctx->recv_mqfd);
+	mq_close(ctx->send_mqfd);
+	ctx->recv_mqfd = 0;
+	ctx->send_mqfd = 0;
+	mq_unlink(mq_name);
+	g_ctx = NULL;
+
+	BLE_LOG_INFO("[BLEMGR] Auto Connection Done\n");
+
+	return NULL;
+}
+
+ble_autocon_result_e ble_manager_autoconnect(ble_client_ctx *ctx)
+{
+	if (ctx->state != BLE_CLIENT_AUTOCONNECTING) {
```

Comment:
How about below?
`if (ctx && ctx->state ...)`

Reply:
@jeongchanKim 
'ctx'  is guaranteed not to be NULL.